test/child/UChildDAI.test.js

```diff
@@ -0,0 +1,151 @@
+import chai from 'chai'
+import chaiAsPromised from 'chai-as-promised'
+import chaiBN from 'chai-bn'
+import BN from 'bn.js'
+import * as sigUtils from 'eth-sig-util'
+import * as ethUtils from 'ethereumjs-util'
+
+import { generateFirstWallets, getSignatureParameters } from '../helpers/utils'
+import { mockValues } from '../helpers/constants'
+import contracts from '../helpers/contracts'
+
+chai
+  .use(chaiAsPromised)
+  .use(chaiBN(BN))
+  .should()
+
+const should = chai.should()
+const wallets = generateFirstWallets({ n: 10 })
+
+contract('UChildDAI', (accounts) => {
+  describe('Set approval using permit function', () => {
+    const cinnamon = 'Cinnamon'
+    const staranise = 'Star anise'
+    const symbol = 'CSC'
+    const decimals = 18
+    const childChainManager = mockValues.addresses[2]
+    const admin = accounts[0]
+    const jack = wallets[4].getAddressString()
+    const jackPK = ethUtils.toBuffer(wallets[4].getPrivateKeyString())
+    const jill = mockValues.addresses[6]
+    const maxAmount = new BN('ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff', 'hex')
+    const zeroAmount = new BN(0)
+    let uChildDAI
+
+    before(async() => {
+      uChildDAI = await contracts.UChildDAI.new({ from: admin })
+      await uChildDAI.initialize(cinnamon, symbol, decimals, childChainManager, { from: admin })
+      await uChildDAI.changeName(staranise, { from: admin })
+    })
+
+    it(`Admin should be able to permit jill to spend jack's tokens using offline signature`, async() => {
+      const name = await uChildDAI.name()
```

Comment:
Can we also assert equal the name, symbol, decimals properties to make sure it is correct?

Reply:
Added assertion for init